makefile.shade

```diff
@@ -929,13 +929,27 @@ functions
             {
                 get
                 {
-                    if (Dependencies.Count > 0)
-                    {
-                        return 1 + Dependencies.Max(d => d.Order);
-                    }
+                    return GetOrder(new HashSet<RepositoryInfo>(), this);
+                }
+            }
 
-                    return 1;
+            private static int GetOrder(HashSet<RepositoryInfo> visited, RepositoryInfo info)
```

Comment:
Though it would be a perf hit, it may be better to have this as IEnumerable to know the path that caused a cyclic dependency. With a hash set, it isn't guaranteed that string.Join will be in added order, and would make the error message more useful.